pkg/canary/scaled_object_reconciler.go

```diff
@@ -206,3 +217,33 @@ func setPrimaryScaledObjectQueries(cd *flaggerv1.Canary, triggers []keda.ScaleTr
 		}
 	}
 }
+
+func makeObjectMetaSo(name string, labels map[string]string, annotations map[string]string, cd *flaggerv1.Canary) metav1.ObjectMeta {
+	return metav1.ObjectMeta{
+		Name:        name,
+		Namespace:   cd.Namespace,
+		Labels:      filterMetadata(labels),
+		Annotations: filterMetadata(annotations),
+		OwnerReferences: []metav1.OwnerReference{
+			*metav1.NewControllerRef(cd, schema.GroupVersionKind{
+				Group:   flaggerv1.SchemeGroupVersion.Group,
+				Version: flaggerv1.SchemeGroupVersion.Version,
+				Kind:    flaggerv1.CanaryKind,
+			}),
+		},
+	}
+}
+
+func initializeTargetSoClone(targetSoClone *keda.ScaledObject) {
```

Comment:
lets update the name to something that indicates that it sets the primary HPA's name

Reply:
Good idea